gemrb/core/Scriptable/Actor.cpp

```diff
@@ -7447,6 +7447,12 @@ void Actor::UpdateActorState(ieDword gameTime) {
 		lastattack = 0;
 	}
 
+	// delayed HP adjustment after max HP modification
+	if (checkHP) {
+		checkHP = 0;
+		pcf_hitpoint(this, 0, BaseStats[IE_HITPOINTS]);
+	}
+
```

Comment:
this is an ugly place to move the check to, did it not work in any of the more natural candidates dealing with effect application?

Reply:
I guess it is.  But the whole idea of this hack was to delay HP check for a tick, and anything in `RefreshEffects()` runs way too often for that.  We _could_ move the check into, say, `RefreshHP()`, but we'd have to manually check if a tick has passed.